hudi-common/src/main/java/org/apache/hudi/common/util/JsonUtils.java

```diff
@@ -29,7 +28,14 @@ public class JsonUtils {
   private static final ObjectMapper MAPPER = new ObjectMapper();
   static {
     MAPPER.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-    MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+    // We need to exclude custom getters, setters and creators which can use member fields
+    // to derive new fields, so that they are not included in the serialization
+    MAPPER.setVisibility(
+        MAPPER.getSerializationConfig().getDefaultVisibilityChecker()
```

Comment:
We can actually do shorter one w/ just `mapper.setVisibility(JsonMethod.FIELD, Visibility.ANY)` and so on